### DISCUSSION TITLE
How to prevent automatic refetch after a mutation? Tried tons of things :)

### DISCUSSION THREAD

**Discussion by sneko** [TanStack/query#5135] [Q&A]
Hi,

In my case a mutation will invalidate an object on the backend so having a new fetch would imply an error changing the UI, and I don't want this.

I tried:
* using the `enabled` on the query, if my boolean is set after the mutation success, the query has already started to fetch
* settings `refetchOnMount: false, refetchOnWindowFocus: false, refetchOnReconnect: false` but it seems it does not apply to post-mutation
* and also `staleTime: Infinity, cacheTime: Infinity` as mentioned by @TkDodo in https://stackoverflow.com/questions/71282427/how-to-fetch-user-details-only-once but it does not prevent the refetch after mutation

I'm a bit loss, what's the magic trick? On my side it would be acceptable otherwise to only fetch once even if it's not ideal.

(I saw a `isMutating` trick to use on `enabled`, it's not what I was expecting since I want to block fetching after a success, but maybe I can deal with that)

Thank you,

EDIT: I'm using `@tanstack/react-query`

**Comment by louis-young** [comment]
Hi 👋 

Please can you provide a minimal reproduction or the relevant code please? If you've configured the query to cache infinitely, it shouldn't refetch unless something else is at play here such as the query key changing.

**Comment by TkDodo** [comment]
> How to prevent automatic refetch after a mutation?

There is no automatic refetch after mutations. It only refetches if you call `invalidateQueries()` after the mutation. Or, it refetches because of something else, like a background refetch if you query is considered stale. It's impossible to say without seeing a reproduction.
